app/actions.py

```diff
@@ -86,9 +86,38 @@ def performDeleteRange(self, upperRow, upperCol, lowerRow, lowerCol):
       self.getText(upperRow, upperCol, lowerRow, lowerCol)))
     self.redo()
 
+  def getBookmarkColor(self):
+    """
+    Returns a new color by cycling through a predefined
+    section of the color palette.
+
+    Args:
+      None.
+
+    Returns:
+      A color (int) for a new bookmark.
+    """
+    if app.prefs.startup['numColors'] == 8:
+      badColorIndices = [0, 7]
+      colorSetSize = 8
+    else:
+      badColorIndices = [103, 106, 107, 111, 121, 122]
+      colorSetSize = 32
+    nextIndex = self.nextBookmarkColorIndex
+    while 1:
+      if nextIndex >= app.prefs.color['bookmarkColorIndexStart'] + colorSetSize:
+        nextIndex = app.prefs.color['bookmarkColorIndexStart']
+      if nextIndex in badColorIndices:
+        nextIndex += 1
+      else:
+        self.nextBookmarkColorIndex = nextIndex + 1
+        return nextIndex
```

Comment:
wdyt of using a goodColorIndices list?
```
goodColorIndices = [104, 105, 108, 109, 110]
if app.prefs.startup['numColors'] == 8:
  goodColorIndices = [1, 2, 3, 4, 5]
self.nextBookmarkColorIndex = (
    self.nextBookmarkColorIndex + 1) % len(goodColorIndices)
return goodColorIndices[self.nextBookmarkColorIndex]
```

Reply:
I guess this would work well, and we can limit it to ~5ish colors each so it would be easier to maintain (so we don't have to rewrite up to 32 colors if we decide on a color scheme change).